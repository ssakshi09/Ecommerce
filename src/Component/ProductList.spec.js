import ProductsList from "./PrdouctsList";
import {
  render,
  fireEvent,
  screen,
  waitForElement,
} from "@testing-library/react";
import "@testing-library/jest-dom/extend-expect";

let handleItems = jest.fn();
const fetchProducts = jest.fn();
describe("ProductList Component", () => {
  const products = [
    {
      id: 1,
      title: "iPhone 9",
      description: "An apple mobile which is nothing like apple",
      price: 549,
      rating: 4.69,
      brand: "Apple",
      images: [
        "https://i.dummyjson.com/data/products/1/1.jpg",
        "https://i.dummyjson.com/data/products/1/2.jpg",
        "https://i.dummyjson.com/data/products/1/3.jpg",
        "https://i.dummyjson.com/data/products/1/4.jpg",
        "https://i.dummyjson.com/data/products/1/thumbnail.jpg",
      ],
    },
  ];

  it("matches snapshot", () => {
    const wrapper = render(
      <ProductsList
        handleItems={handleItems}
        fetchProducts={fetchProducts}
        products={products}
      />
    );

    expect(wrapper).toMatchSnapshot();
  });

  it("should call the handleItems function when the add to cart button is clicked", () => {
    const wrapper = render(
      <ProductsList
        handleItems={handleItems}
        fetchProducts={fetchProducts}
        products={products}
      />
    );
    const addButton = wrapper.getAllByTestId("add-to-cart");
    addButton.forEach((button) => {
      fireEvent.click(button);
      expect(handleItems).toHaveBeenCalled();
    });
  });

  it("should call the handleItems function when the add to cart button is clicked", () => {
    const wrapper = render(
      <ProductsList
        handleItems={handleItems}
        fetchProducts={fetchProducts}
        products={products}
      />
    );
    const addButton = wrapper.getAllByTestId("show-details-button");
    // addButton.forEach((button) => {
    expect(addButton[0]).toHaveTextContent("show details");
    fireEvent.click(addButton[0]);
    //expect(handleShowDetails)
    expect(addButton[0]).toHaveTextContent("hide details");
    // });
  });

  // it(" should have been called fetchproducts", () => {
  //   const wrapper = render(
  //     <ProductsList
  //       handleItems={handleItems}
  //       fetchProducts={fetchProducts}
  //       products={products}
  //     />
  //   );
  //   expect(fetchProducts).toHaveBeenCalled();
  // });

  it("should render the add-to-cart button", () => {
    const products = [
      {
        id: 1,
        title: "iPhone 9",
        description: "An apple mobile which is nothing like apple",
        price: 549,
        rating: 4.69,
        brand: "Apple",
        images: [
          "https://i.dummyjson.com/data/products/1/1.jpg",
          "https://i.dummyjson.com/data/products/1/2.jpg",
          "https://i.dummyjson.com/data/products/1/3.jpg",
          "https://i.dummyjson.com/data/products/1/4.jpg",
          "https://i.dummyjson.com/data/products/1/thumbnail.jpg",
        ],
      },
    ];
    const { getAllByTestId } = render(
      <ProductsList
        handleItems={handleItems}
        fetchProducts={fetchProducts}
        products={products}
      />
    );

    expect(getAllByTestId("add-to-cart")).toBeTruthy();
  });
});
