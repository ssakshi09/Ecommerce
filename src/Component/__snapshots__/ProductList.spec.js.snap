// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ProductList Component matches snapshot 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <div
        class="makeStyles-root-1"
        data-testid="product"
      >
        <div
          class="MuiGrid-root MuiGrid-container MuiGrid-spacing-xs-3"
        >
          <div
            class="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-4"
          >
            <div
              class="MuiPaper-root makeStyles-paper-2 MuiPaper-elevation1 MuiPaper-rounded"
            >
              <img
                loading="lazy"
                src="https://i.dummyjson.com/data/products/1/1.jpg"
                srcset="https://i.dummyjson.com/data/products/1/1.jpg?w=164&h=164&fit=crop&auto=format&dpr=2 2x"
              />
              <h3>
                iPhone 9
              </h3>
              <p>
                $
                549
              </p>
              <div>
                <button
                  class="MuiButtonBase-root MuiButton-root MuiButton-text"
                  data-testid="show-details-button"
                  tabindex="0"
                  type="button"
                >
                  <span
                    class="MuiButton-label"
                  >
                    show details
                  </span>
                  <span
                    class="MuiTouchRipple-root"
                  />
                </button>
              </div>
              <button
                class="MuiButtonBase-root MuiButton-root MuiButton-text"
                data-testid="add-to-cart"
                tabindex="0"
                type="button"
              >
                <span
                  class="MuiButton-label"
                >
                  Add to Cart
                </span>
                <span
                  class="MuiTouchRipple-root"
                />
              </button>
            </div>
          </div>
        </div>
      </div>
    </div>
  </body>,
  "container": <div>
    <div
      class="makeStyles-root-1"
      data-testid="product"
    >
      <div
        class="MuiGrid-root MuiGrid-container MuiGrid-spacing-xs-3"
      >
        <div
          class="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-4"
        >
          <div
            class="MuiPaper-root makeStyles-paper-2 MuiPaper-elevation1 MuiPaper-rounded"
          >
            <img
              loading="lazy"
              src="https://i.dummyjson.com/data/products/1/1.jpg"
              srcset="https://i.dummyjson.com/data/products/1/1.jpg?w=164&h=164&fit=crop&auto=format&dpr=2 2x"
            />
            <h3>
              iPhone 9
            </h3>
            <p>
              $
              549
            </p>
            <div>
              <button
                class="MuiButtonBase-root MuiButton-root MuiButton-text"
                data-testid="show-details-button"
                tabindex="0"
                type="button"
              >
                <span
                  class="MuiButton-label"
                >
                  show details
                </span>
                <span
                  class="MuiTouchRipple-root"
                />
              </button>
            </div>
            <button
              class="MuiButtonBase-root MuiButton-root MuiButton-text"
              data-testid="add-to-cart"
              tabindex="0"
              type="button"
            >
              <span
                class="MuiButton-label"
              >
                Add to Cart
              </span>
              <span
                class="MuiTouchRipple-root"
              />
            </button>
          </div>
        </div>
      </div>
    </div>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
